modules/core/src/main/java/org/apache/synapse/api/ApiUtils.java

```diff
@@ -158,6 +165,89 @@ public static String getSubRequestPath(MessageContext synCtx) {
         return (String) synCtx.getProperty(RESTConstants.REST_SUB_REQUEST_PATH);
     }
 
+    /**
+     * Checks whether the provided resource is capable of processing the message from the provided message context.
+     * The resource becomes capable to do this when the it contains either the name of the api caller,
+     * or {@value ApiConstants#DEFAULT_BINDING_ENDPOINT_NAME}, in its binds-to.
+     *
+     * @param resource  Resource object
+     * @param synCtx    MessageContext object
+     * @return          Whether the provided resource is bound to the provided message context
+     */
+    public static boolean isBound(Resource resource, MessageContext synCtx) {
+        Collection<String> bindings = resource.getBindsTo();
+        Object apiCaller = synCtx.getProperty(ApiConstants.API_CALLER);
+        if (apiCaller != null) {
+            return bindings.contains(apiCaller.toString());
+        }
+        return bindings.contains(ApiConstants.DEFAULT_BINDING_ENDPOINT_NAME);
+    }
+
+    public static Set<Resource> getAcceptableResources(Map<String, Resource> resources, MessageContext synCtx) {
+        Set<Resource> acceptableResources = new LinkedHashSet<Resource>();
+        for (Resource r : resources.values()) {
+            if (isBound(r, synCtx) && r.canProcess(synCtx)) {
+                acceptableResources.add(r);
+            }
+        }
+        return acceptableResources;
+    }
+
+    public static API getSelectedAPI(MessageContext synCtx) {
+        API selectedApi;
+        //getting the API collection from the synapse configuration to find the invoked API
+        Collection<API> apiSet = synCtx.getEnvironment().getSynapseConfiguration().getAPIs();
+        //Since swapping elements are not possible with sets, Collection is converted to a List
+        List<API> defaultStrategyApiSet = new ArrayList<API>(apiSet);
+        //To avoid apiSet being modified concurrently
+        List<API> duplicateApiSet = new ArrayList<>(apiSet);
+        //identiy the api using canProcess method
+        for (API api : duplicateApiSet) {
+            if (identifySelectedAPI(api, synCtx, defaultStrategyApiSet)) {
+                selectedApi = api;
+                return selectedApi;
+            }
+        }
+        for (API api : defaultStrategyApiSet) {
+            api.setLogSetterValue();
+            if (api.canProcess(synCtx)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Located specific API: " + api.getName() + " for processing message");
+                }
+                selectedApi = api;
+                return selectedApi;
+            }
+        }
+        return null;
+    }
+
+    public static boolean identifySelectedAPI(API api, MessageContext synCtx, List defaultStrategyApiSet) {
```

Comment:
do we have to make this public? what's the intension of this?

Reply:
Yes we can make it private